pkg/factory/factory.go

```diff
@@ -0,0 +1,33 @@
+package factory
+
+import (
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/OctopusDeploy/cli/pkg/apiclient"
+	"github.com/OctopusDeploy/cli/pkg/question"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/client"
+)
+
+type factory struct {
+	client apiclient.ClientFactory
+	asker  question.Asker
```

Comment:
Happy to see you've gone with a dependency injection based approach rather than a global :-)